### PR TITLE
Implemenet cumsum

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `cumsum` method. It currently doesn't support `axis` argument.

### CODE REVIEW

File: cupy/math/sumprod.py [hunk 1]
```diff
@@ -46,7 +51,40 @@ def prod(a, axis=None, dtype=None, out=None, keepdims=False):
...
-# TODO(okuta): Implement cumsum
+def cumsum(a, axis=None, dtype=None, out=None):
+    if axis is None:
+        a = a.ravel()
+    else:
+        raise ValueError("'axis' option is not supported")
+
+    if out is None:
+        if dtype is None:
+            if a.dtype in ['?', 'b', 'h', 'i']:
```
Comment (gwtnb) [inline cupy/math/sumprod.py]: `'l'` is 32bit on Windows.

File: cupy/math/sumprod.py [hunk 2]
```diff
@@ -46,7 +51,40 @@ def prod(a, axis=None, dtype=None, out=None, keepdims=False):
...
+                dtype = numpy.dtype(numpy.intp)
+            elif a.dtype in ['B', 'H', 'I']:
```
Comment (gwtnb) [inline cupy/math/sumprod.py]: `L` is 32bit on Windows.

**Comment by unnonouno** [comment]
I fixed this code based on numpy's code.
https://github.com/numpy/numpy/blob/v1.12.0/numpy/core/src/umath/ufunc_object.c#L4052

**Comment by gwtnb** [comment]
LGTM

**Pull request merged**